### ISSUE TITLE
jest-cucumber doesn't clear DOM after each Test suite when using it with @testing-library

### ISSUE THREAD

**Issue by flanzana** [bencompton/jest-cucumber#90]
Hi! 

I wanted to use `jest-cucumber` with @testing-library but I have come across a bug.

__________
TL;DR: `jest-cucumber` doesn't clear DOM after each Test suite when using it with @testing-library React.
___________

I mock a different response from fetching data for each scenario. Here is my [steps file](https://github.com/flanzana/test-cucumber/blob/master/src/invoices/cucumber/Invoices.steps.tsx) as an example. Somehow the first scenario renders with resolved data from both scenarios, while the second scenario doesn't render anything (it has empty `<body />`). The problem is that body/DOM is not cleared after each test suite, even if I tried to use `cleanup()` from TLR or simple `document.body.innerHTML = ''` . I was trying to find a reason why it doesn't work for the whole week, but nothing works. I was searching through GitHub to find some other repos which use `jest-cucumber` with @testing-library, but no one is mocking responses as I would like to. 😞

I tried the same approach with the `react-test-renderer` (in [file](https://github.com/flanzana/test-cucumber/blob/master/src/invoices/cucumber/Invoices2.steps.tsx)) and it works well, but as we are already using @testing-library for unit tests, we would like to use it as well for Gherkin steps instead of installing `react-test-renderer` just for steps.

I believe that the problem shouldn’t be in @testing-library, because if I do just unit tests without Gherkin, the body is cleared and it works fine for each test suite ([file](https://github.com/flanzana/test-cucumber/blob/master/src/invoices/components/__tests__/InvoicesContainer.test.tsx) of unit tests).

Can someone take a look at the issue, please? 🙏 🙏

**Comment by TheSavageDev** [comment]
Have you found a solution for this?

**Comment by TheSavageDev** [comment]
I may have found one.

```js
beforeEach(() => {
  render(<Component {...props} />);
});
afterEach(() => {
  cleanup();
});
```

Then in your tests use `screen.getByTextId()` Seems to be working for me.

**Comment by labibimamhidayat** [comment]
hello @flanzana have you found something to solve this issue?

**Comment by labibimamhidayat** [comment]
> I may have found one.
> 
> ```js
> beforeEach(() => {
>   render(<Component {...props} />);
> });
> afterEach(() => {
>   cleanup();
> });
> ```
> 
> Then in your tests use `screen.getByTextId()` Seems to be working for me.

hello, imho its not the best solution to resolve this issue, because according eslint/testing-library which i used, it clearly suggest not to call render inside before each `Forbidden usage of `render` within testing framework `beforeEach` setupeslinttesting-library/no-render-in-setup`

**Comment by fraindz** [comment]
This is actually an issue. Each render inside of jest is rendered on top of the previous render. The previous DOM is not cleared.
